pkg/skaffold/build/local/bazel.go

```diff
@@ -43,20 +49,52 @@ func (b *Builder) buildBazel(ctx context.Context, out io.Writer, workspace strin
 		return "", errors.Wrap(err, "running command")
 	}
 
-	bazelBin, err := bazelBin(ctx, workspace, a)
+	bazelBin, err := bazelBin(ctx, workspace, a.BazelArtifact)
 	if err != nil {
 		return "", errors.Wrap(err, "getting path of bazel-bin")
 	}
 
-	tarPath := buildTarPath(a.BuildTarget)
-	imageTar, err := os.Open(filepath.Join(bazelBin, tarPath))
+	tarPath := filepath.Join(bazelBin, buildTarPath(a.BazelArtifact.BuildTarget))
+
+	if b.pushImages {
+		uniqueTag := a.ImageName + ":" + util.RandomID()
+		return pushImage(tarPath, uniqueTag)
```

Comment:
would there ever be a situation where we'd want to load the image to the local daemon as well as build straight to the registry? might be worth considering making that configurable, though i'm not sure how it would look. just something to think about